collect/collect.go

```diff
@@ -617,15 +617,18 @@ func mergeTraceAndSpanSampleRates(sp *types.Span, traceSampleRate uint, dryRunMo
 	}
 }
 
-func isRootSpan(sp *types.Span) bool {
-	parentID := sp.Data["trace.parent_id"]
-	if parentID == nil {
-		parentID = sp.Data["parentId"]
-		if parentID == nil {
-			// no parent ID present; it's a root span
-			return true
+func isRootSpan(sp *types.Span, c config.Config ) bool {
+	var parentID string
+	for _, ParentSpanId := range c.GetParentIdFieldNames() {
+		if pID, ok := sp.Data[ParentSpanId]; ok {
+			parentID = pID.(string)
```

Comment:
Why are you asserting it as a string? Seems like the nil test here should work -- if not, why not? This is a semantic change so I'd want to be sure it's correct.

Reply:
The new code seems to be working and no longer forces the string type in the function